### ISSUE TITLE
Accessory Display Issue. (iOS 13 Grouping) 

### ISSUE THREAD

**Issue by iRayanKhan** [aarons22/homebridge-bond-v2#4]
Hello, I was looking at the config part of the ReadME and noticed it needed bond ID and token, no big deal right? I checked the screenshot and it said you were running the V2.x firmware, my bond is running v1.138.1, when I tried to update it said I'm on the latest. Any guesses on how to update? Cheers

**Comment by iRayanKhan** [comment]
Update: I was using the old bond app, it appears there is a new one called "Bond Home" in which you can upgrade the firmware to V2.x.

**Comment by iRayanKhan** [comment]
I now get this issue:
```
Unhandled rejection StatusCodeError: 401 - undefined
    at new StatusCodeError (/usr/lib/node_modules/homebridge-bond-v2/node_modules/request-promise-core/lib/errors.js:32:15)
    at Request.plumbing.callback (/usr/lib/node_modules/homebridge-bond-v2/node_modules/request-promise-core/lib/plumbing.js:104:33)
    at Request.RP$callback [as _callback] (/usr/lib/node_modules/homebridge-bond-v2/node_modules/request-promise-core/lib/plumbing.js:46:31)
    at Request.self.callback (/usr/lib/node_modules/homebridge-bond-v2/node_modules/request/request.js:185:22)
    at Request.emit (events.js:193:13)
    at Request.<anonymous> (/usr/lib/node_modules/homebridge-bond-v2/node_modules/request/request.js:1161:10)
    at Request.emit (events.js:193:13)
    at IncomingMessage.<anonymous> (/usr/lib/node_modules/homebridge-bond-v2/node_modules/request/request.js:1083:12)
    at Object.onceWrapper (events.js:281:20)
    at IncomingMessage.emit (events.js:198:15)
    at endReadableNT (_stream_readable.js:1139:12)
    at processTicksAndRejections (internal/process/task_queues.js:81:17)

```

**Comment by iRayanKhan** [comment]
Another issue is, if you run more than 1 home bridge instance, you need the config parameter is every config.json. I got away with this in the old plugin by using a separate bond account with no devices so it added nothing, this plugin however forces home bridge to quit is any info is wrong.
```
/usr/lib/node_modules/homebridge-bond-v2/dist/BondApi.js:14
        this.bondToken = config.bond_token;
                                ^

TypeError: Cannot read property 'bond_token' of null
    at new BondApi (/usr/lib/node_modules/homebridge-bond-v2/dist/BondApi.js:14:33)
    at new BondPlatform (/usr/lib/node_modules/homebridge-bond-v2/dist/index.js:23:24)
    at Server._loadDynamicPlatforms (/usr/lib/node_modules/homebridge/lib/server.js:347:30)
    at Server.run (/usr/lib/node_modules/homebridge/lib/server.js:88:8)
    at module.exports (/usr/lib/node_modules/homebridge/lib/cli.js:45:10)
    at Object.<anonymous> (/usr/lib/node_modules/homebridge/bin/homebridge:17:22)
    at Module._compile (internal/modules/cjs/loader.js:816:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:827:10)
    at Module.load (internal/modules/cjs/loader.js:685:32)
    at Function.Module._load (internal/modules/cjs/loader.js:620:12)
    at Function.Module.runMain (internal/modules/cjs/loader.js:877:12)
    at internal/main/run_main_module.js:21:11
```

**Comment by aarons22** [comment]
A couple things:
- update to the latest version (0.2.0), which is now using the IP Address of the Bond instead of the BondID. That might fix your 401 issue. Let me know!
- I hadn’t thought about it causing issues with multiple instances of homebridge. Is there a good example of a homebridge plugin that handles this well? Personally I’ve always just uninstalled a plugin if it wasn’t being used so I’ve never seen this before.

**Comment by iRayanKhan** [comment]
Hello thanks for the reply,
I've updated the firmware before I installed the plugin so this caused post-install. A plugin that handles it well is [MagicHome]https://github.com/sahilchaddha/homebridge-magichome-platform

**Comment by iRayanKhan** [comment]
And some may be wondering why I use multiple instances, the truth is, a plugin may decide to fail and may cripple my access to get into my via Garage or not being able to see my Cameras.

**Comment by iRayanKhan** [comment]
Also this is my config:
```
{
	"bridge": {
		"name": "Homebridge",
		"username": "CC:22:3D:E3:CE:30",
		"port": 51826,
		"pin": "031-45-154"
	},
	"description": "Homebridge",
	"accessories": [{
			"accessory": "DummySwitch",
			"name": "Living Room Switch",
			"stateful": true
		},
		{
			"accessory": "DummySwitch",
			"name": "MasterBedroom Switch",
			"stateful": true
		},
		{
			"accessory": "RaspberryPiTemperature",
			"name": "CPU",
			"updateInterval": 1000
		}

	],
	"platforms": [{
			"platform": "BondHome",
			"bond_ip_address": "10.0.0.77",
			"bond_token": "Removed for Privacy"
		}, {
			"platform": "MagicHome-Platform",
			"debug": true,
			"lights": [{
					"name": "Kitchen LED Strip1",
					"ip": "10.0.0.81",
					"setup": "RGBW",
					"purewhite": false
				},
				{
					"name": "Kitchen LED Strip2",
					"ip": "10.0.0.54",
					"setup": "RGBW",
					"purewhite": false
				},
				{
					"name": "Kitchen LED Strip3",
					"ip": "10.0.0.50",
					"setup": "RGBW",
					"purewhite": false
				},
				{
					"name": "Game Room Light Strip",
					"ip": "10.0.0.11",
					"setup": "RGBW",
					"purewhite": false
				}
			]
		},
		{
			"platform": "MagicHome-Platform",
			"debug": true,
			"lights": [{
					"name": "Kitchen LED Strip1",
					"ip": "10.0.0.81",
					"setup": "RGBW",
					"purewhite": false
				},
				{
					"name": "Kitchen LED Strip2",
					"ip": "10.0.0.54",
					"setup": "RGBW",
					"purewhite": false
				},
				{
					"name": "Kitchen LED Strip3",
					"ip": "10.0.0.50",
					"setup": "RGBW",
					"purewhite": false
				},
				{
					"name": "Game Room Light Strip",
					"ip": "10.0.0.11",
					"setup": "RGBW",
					"purewhite": false
				}
			]

		}
	]
}
```

**Comment by aarons22** [comment]
Makes sense. I’ll take a look soon!

**Comment by aarons22** [comment]
@iRayanKhan I was able to make the plugin silently fail with a log error instead of completely killing the homebridge process. I'm not crazy about the solution, so we'll see if this is the approach  I stick with long term. 

As for your 401 issue, I think your BOND_TOKEN is incorrect. I was able to replicate `Unhandled rejection StatusCodeError: 401 - undefined` by providing the plugin with a bad token. Can you double check your token?

**Comment by aarons22** [comment]
can you update to 0.2.1 and try again?

**Comment by iRayanKhan** [comment]
Will try it as soon as I get home.

**Comment by iRayanKhan** [comment]
It appears the 401 has been resolved, my bond local token changed it seems. Every works great, except in iOS 13 is displays as a Light rather than a Fan. (Screenshots) 
![IMG_0844](https://user-images.githubusercontent.com/34495712/65465856-f0b29300-de22-11e9-9cce-c61e25c6a2f8.PNG)
![IMG_0843](https://user-images.githubusercontent.com/34495712/65465857-f0b29300-de22-11e9-91b7-f699b602951b.PNG)

**Comment by jimboslicethat** [comment]
I'm seeing this issue as well, though I assumed that it was in issue with Homekit in iOS13 that would be patched in `13.1`.

**Comment by iRayanKhan** [comment]
Update: So the old bond program had the Fan Accessory display first, then Switch (fan Reset), then at the bottom the Light. So I guess a patch for this would be to have it initialize the Fan First, light last. But thinking about it, they initialize together due to the new Grouping Feature.

**Comment by aarons22** [comment]
I just tested it out and it appears the grouping does depend on the order you add it. This is now fixed in 0.2.2

**Comment by aarons22** [comment]
@iRayanKhan also, I was able to knock out rotation direction switch just now, which is included in 0.3.0!

**Comment by iRayanKhan** [comment]
Glad to say, it works!. Thank you so much for this plugin it really has fixed all the issues I've had with the V1 plugin.
For anyone who installed this plugin before V3 and it still groups wrong after updating here is what to do.
Update Plugin: Sudo nom I -g Homebridge-bond-v2
Clear you're home bridges Cached Accessories (DO NOT DELETE PERSIST) : cd (your Homebridge dir)
rm -r Accessories
Then restart Homebridge if you use PM2 then do pm2 restart homebridge
Then reload the Home app and you'll see the fans re-appear at the bottom of the screen and you can move them where you like. If you try to change the Fan's icon and it doesn't let you hit "Accessories" and you can individually set the Fan and Light's accessory Name and Icon.
